travis.sh

```diff
@@ -12,3 +12,25 @@ make -f posix.mak all DMD="$(which dmd)"
 make -f posix.mak test DMD="$(which dmd)" \
     RDMD_TEST_COMPILERS=dmd,"$LDMD2" \
     VERBOSE_RDMD_TEST=1
+
+# Test setup.sh
+shellcheck setup.sh
+
+dmd=dmd/generated/linux/release/64/dmd
+dir=generated/setup.sh-test
+cwd="$(pwd)"
+
+# check initial checkout
+rm -rf "$dir" && mkdir "$dir" && pushd "$dir"
+echo "y" | "$cwd"/setup.sh
+echo 'void main(){ import std.stdio; "Hello World".writeln;}' | "./${dmd}" -run - | grep -q "Hello World"
+
+# test updates
+echo "y" | "$cwd"/setup.sh
+echo 'void main(){ import std.stdio; "Hello World".writeln;}' | "./${dmd}" -run - | grep -q "Hello World"
+popd && rm -rf "$dir" && mkdir "$dir" && pushd "$dir"
+
+# test checking out tags
+echo "y" | "$cwd"/setup.sh --tag=2.078.1
+echo 'void main(){ import std.stdio; __VERSION__.writeln;}' | "./2.078.1/${dmd}" -run - | grep -q "2078"
+popd
```

Comment:
As I understand, this will attempt to build the current master of the main D components. One issue with that is that failures (or changes leading this script to fail) there will propagate here, so we will see things like master being green, but a PR being red due to no fault of any change in the PR.

Reply:
1) This here checks out a specific tag - it will always be the same
2) For checking out master (above) - imho it's more than safe to assume that the current hello world will compile forever. I agree that our Makefiles or something else might change though.
So yes it has this potential downsides, but
- I don't think that `import std.stdio; "Hello World".writeln;` will break in the foreseeable future (it's in a lot of testsuites of our CIs)
- it's better than the status quo of no notification when it breaks
3) Yeah, ideally we do such a test on the project tester, but one step at a time ;-)